### ISSUE TITLE
Energy Storage Int.MaxValue bug with negative energy

### ISSUE THREAD

**Issue by TheDeaX-Lab** [brandon3055/Draconic-Evolution#1455]
## Base information
* Minecraft version: 1.12.2
* Mod version: 2.3.25.351
* Minecraft Forge version: 14.23.5.2847
* Mod Pack: custom

## Crash report
no crashes

## Video with bug
https://youtu.be/HPXVK8LfKFc

tested with rainbow generator in mod ExtraUtilities

**Comment by FoxMcloud5655** [comment]
Are you playing using 32-bit Java or 64-bit?

**Comment by TheDeaX-Lab** [comment]
> Are you playing using 32-bit Java or 64-bit?

This I cannot say for sure, I will ask the server administrator. And so I have a screenshot of a discharged sword less than -2.147B
https://media.discordapp.net/attachments/465536899637968900/750290166249160764/2020-08-27_14.17.36.png?width=1278&height=677

**Comment by TheDeaX-Lab** [comment]
@FoxMcloud5655 Confirmed by the server administrator, 64 bit java is installed with 12 gigabytes of RAM

**Comment by FoxMcloud5655** [comment]
Hmm...  Interesting for sure.  A world download and a list of mods would be extremely helpful, but seeing as you're playing on a server, I don't suppose the world download is going to be possible.

If you wouldn't mind providing a list of mods, that should help to narrow it down, in case it is a mod conflict.  Seeing as how brandon3055 is currently working on the port to 1.15, I don't think this is going to be tackled anytime soon.  If I get some time, however, I'll look at it for sure, and see if I can submit a pull request if it's DE's fault.

**Comment by TheDeaX-Lab** [comment]
ok, here is a list of mods in this server, if that helps. The card will take a long time to load, it takes about 5 GB

**Comment by TheDeaX-Lab** [comment]
@FoxMcloud5655 I still have guesses about what CoFH fails

**Comment by TheDeaX-Lab** [comment]
as a reference, I compare with 1.7.10 everything was perfect there, and your self-written storage library IExtendedRFStorage.java was used in which there were double and long number types

**Comment by TheDeaX-Lab** [comment]
@FoxMcloud5655  Are there any guesses about this? It's been 1 week

**Comment by FoxMcloud5655** [comment]
Currently looking into it now.  I've been trying to reproduce the issue with just Draconic Evolution installed, and everything works perfectly fine.  Tried many methods of charging and discharging.

I'm going to go ahead and use your mods and world to see if this can be reproduced on my machine.  If I can't, then it's a server problem.  If I can, well, we'll see if this bug is something we can fix (or even track down) or not.

**Comment by FoxMcloud5655** [comment]
Well, good news, I guess.  I accidentally ran your modpack here with 4 GB of RAM, and it actually loaded up perfectly fine.  So there's that, I suppose :P

The forced multiplayer join makes it hard to boot up a single player game, and it's all in Russian, so I had to guess at what everything does.  A minor setback, though.  Just had to remove the mod responsible.

Had to explore to find the core...  Gotta say though, I love that church at spawn.  Very neat.  I found your problem.

The Spectre Energy Injector.

This is not a problem of Draconic Evolution, but of Random Things.

https://www.reddit.com/r/feedthebeast/comments/ggoz6q/weird_problem_with_random_things_spectre_energy/
